packages/@lwc/engine-dom/src/renderer-factory.ts

```diff
@@ -0,0 +1,386 @@
+        (element as HTMLElement | SVGElement).style.setProperty(
+            name,
+            value,
+            important ? 'important' : ''
+        );
+    }
+
+    function getBoundingClientRect(element: Element): DOMRect {
+        return element.getBoundingClientRect();
+    }
+
+    function querySelector(element: Element, selectors: string): Element | null {
+        return element.querySelector(selectors);
+    }
+
+    function querySelectorAll(element: Element, selectors: string): NodeList {
+        return element.querySelectorAll(selectors);
+    }
+
+    function getElementsByTagName(element: Element, tagNameOrWildCard: string): HTMLCollection {
+        return element.getElementsByTagName(tagNameOrWildCard);
+    }
+
+    function getElementsByClassName(element: Element, names: string): HTMLCollection {
+        return element.getElementsByClassName(names);
+    }
+
+    function getChildren(element: Element): HTMLCollection {
+        return element.children;
+    }
+
+    function getChildNodes(element: Element): NodeList {
+        return element.childNodes;
+    }
+
+    function getFirstChild(element: Element): Node | null {
+        return element.firstChild;
+    }
+
+    function getFirstElementChild(element: Element): Element | null {
+        return element.firstElementChild;
+    }
+
+    function getLastChild(element: Element): Node | null {
+        return element.lastChild;
+    }
+
+    function getLastElementChild(element: Element): Element | null {
+        return element.lastElementChild;
+    }
+
+    function isConnected(node: Node): boolean {
+        return node.isConnected;
+    }
+
+    function assertInstanceOfHTMLElement(elm: any, msg: string) {
+        assertInvariant(elm instanceof HTMLElement, msg);
+    }
+
+    const HTMLElementExported = HTMLElementConstructor as typeof HTMLElement;
+
+    const renderer = {
+        HTMLElementExported,
+        insert,
+        remove,
+        cloneNode,
+        createFragment,
+        createElement,
+        createText,
+        createComment,
+        nextSibling,
+        attachShadow,
+        getProperty,
+        setProperty,
+        setText,
+        getAttribute,
+        setAttribute,
+        removeAttribute,
+        addEventListener,
+        removeEventListener,
+        dispatchEvent,
+        getClassList,
+        setCSSStyleProperty,
+        getBoundingClientRect,
+        querySelector,
+        querySelectorAll,
+        getElementsByTagName,
+        getElementsByClassName,
+        getChildren,
+        getChildNodes,
+        getFirstChild,
+        getFirstElementChild,
+        getLastChild,
+        getLastElementChild,
+        isConnected,
+        assertInstanceOfHTMLElement,
+        defineCustomElement,
+        getCustomElement,
+    };
+    // Meant to inherit any properties passed via the base renderer as the argument to the factory.
+    Object.setPrototypeOf(renderer, baseRenderer);
```

Comment:
Can you explain why we are doing the inheritance here? 

From what I understand, locker should be able to override these methods in the renderer with lockerized versions of those (which are in `baseRenderer`). However based on this line, The methods in `renderer` will get invoked rather than the ones in `baseRenderer`.